daemon/endpoint.go

```diff
@@ -264,7 +265,7 @@ func (h *patchEndpointID) Handle(params PatchEndpointIDParams) middleware.Respon
 			string(epTemplate.State) == endpoint.StateReady) &&
 		ep.GetStateLocked() != endpoint.StateWaitingForIdentity {
 		// Will not change state if the current state does not allow the transition.
-		if ep.SetStateLocked(endpoint.StateWaitingForIdentity) {
+		if ep.SetStateLocked(endpoint.StateWaitingForIdentity, "Update endpoint from API PATCH") {
```

Comment:
I think several of these reasons should become more end user friendly going forward but I understand the desire to provide specific context and reasoning so I'm fine to start with this and eventually add a more end user focused reason summary later.